src/main/java/ch/uzh/ifi/hase/soprafs21/controller/UsersApiController.java

```diff
@@ -175,6 +171,20 @@ public UsersApiController(NativeWebRequest request, UserService userService, Cha
                 .body(userLoginPostDto);
     }
 
-
-
+    @Override
+    public ResponseEntity<UserDto> usersUserIdGet(Long userId) throws Exception {
+        UserDto userDto = null;
+        if (userId != null) {
+            userDto = userService.getUserDetails(userId);
+            if(null == userDto){
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found");
+            }
+        }
+        else {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "Invalid input userId cannot be null");
+        }
```

Comment:
More a matter of style, but it might be nicer to catch these special cases in a single if and throw rather than nesting like this for readability:

if userId == null:
  throw bad request
if userDto == null:
  throw not found

...return

Reply:
updated